net/peer.go

```diff
@@ -804,6 +808,30 @@ func (p *Peer) AddP2PCollections(collections []string) error {
 		addedTopics = append(addedTopics, col)
 	}
 
+	// If adding the collection topics succeeds, we remove the collections' documents
+	// from the pubsub topics to avoid receiving duplicate events.
+	for _, col := range collections {
```

Comment:
thought: This is potentially very expensive - how long do you expect this code to live?  It might be worth noting this in the public documentation.

Reply:
At the moment I can't think of a better way to handle this. It would only be expensive if a dev decides to subscribe to a given collection after it has millions of documents. It's also a one-off event if someone was to do it so I don't think we need to worry about speed too much. Not at this time anyways.

I can add a warning to the doc though.